kedro/runner/runner.py

```diff
@@ -380,7 +398,7 @@ def _find_initial_node_group(pipeline: Pipeline, nodes: Iterable[Node]) -> list[
 
 def run_node(
     node: Node,
-    catalog: DataCatalog,
+    catalog: DataCatalog | KedroDataCatalog,
```

Comment:
If this is for the PoC, could we (just temporarily) come up with something like:
```python
type AbstractDataCatalog = DataCatalog | KedroDataCatalog
```

And then simply use that everywhere until we design the `AbstractDataCatalog`?